lib/kernel/src/code.erl

```diff
@@ -506,7 +511,13 @@ ensure_modules_loaded_1(Ms0) ->
 		 ok -> Error0;
 		 {error,Err} -> Err ++ Error0
 	     end,
-    ensure_modules_loaded_2(OnLoad, Error1).
+    case get_mode() of
+        interactive ->
+            ensure_modules_loaded_2(OnLoad, Error1);
+        embedded ->
+            Error2 = [{M, {error, embedded}} || {M, _} <- OnLoad] ++ Error1,
+            ensure_modules_loaded_2([], Error2)
```

Comment:
This would just return `{error, Error2}` right?

Reply:
We may not have any `OnLoad` modules on partition, which means that the line above does not add new errors and we call `ensure_modules_loaded_2([], [])`. :)